lib/features/details_page/details_page.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:flutter/material.dart';
+import 'package:pokedex/api/models/pokemon_model.dart';
+import 'package:pokedex/api/models/pokemon_type_model.dart';
+import 'package:pokedex/features/about_tab/about_tab_connector.dart';
+import 'package:pokedex/features/base_stats_tab/base_stats_tab_connector.dart';
+import 'package:pokedex/features/evolution_tab/evolution_tab_connector.dart';
+import 'package:pokedex/features/home_page/widgets/type.dart';
+import 'package:pokedex/features/moves_tab/moves_tab_connector.dart';
+import 'package:pokedex/utilities/colors.dart';
+import 'package:pokedex/utilities/constants.dart';
+import 'package:pokedex/utilities/extensions.dart';
+import 'package:pokedex/utilities/string_constants.dart';
+
+class DetailsPage extends StatelessWidget {
+  const DetailsPage({
+    this.specificPokemon,
+    this.types,
+  });
+
+  final PokemonModel? specificPokemon;
+  final List<PokemonTypeModel>? types;
+
+  @override
+  Widget build(BuildContext context) {
+    final textTheme = Theme.of(context).textTheme;
+
+    return (types != null)
+        ? Scaffold(
+            backgroundColor: types?.first.name.toString().pokemonColor,
+            appBar: AppBar(backgroundColor: red),
+            body: Center(
+              child: Column(
+                children: [
+                  Padding(
+                      padding: const EdgeInsets.all(12.0),
+                      child: Column(
+                        crossAxisAlignment: CrossAxisAlignment.start,
+                        children: [
+                          Text(
+                            specificPokemon?.name.toString().replaceDash.toTitleCase ?? '',
+                            style: textTheme.headline1,
+                          ),
+                          Align(
+                            alignment: Alignment.centerRight,
+                            child: Text(
+                              specificPokemon?.url?.pokemonId.toString().formatID ?? '',
+                              style: textTheme.headline1?.copyWith(fontSize: 16),
+                            ),
+                          ),
+                          Row(
+                            children: [...?types?.map((type) => Type(typeName: type.name))],
+                          ),
+                          Center(
+                            child: CachedNetworkImage(
+                              imageUrl: specificPokemon?.url?.pokemonId.toString().pokemonImage ?? '',
```

Comment:
Same here, move `pokemonImage` extension to `int `extensions then remove `.toString()`